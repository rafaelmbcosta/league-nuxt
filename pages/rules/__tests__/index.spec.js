/* global jest */

import { shallowMount, createLocalVue } from "@vue/test-utils"
import Vuetify from "vuetify"
import VueApollo from "vue-apollo"
import { createMockClient } from "mock-apollo-client"
import index from "../index"
import CURRENT_RULES_QUERY from "@/graphql/queries/currentRules.gql"

const localVue = createLocalVue()
localVue.use(VueApollo)
const vuetify = new Vuetify()

describe("index", () => {
  let wrapper
  let mockClient
  let apolloProvider
  let requestHandlers

  const createComponent = (handlers) => {
    mockClient = createMockClient()
    apolloProvider = new VueApollo({
      defaultClient: mockClient
    })

    requestHandlers = {
      currentRulesHandler: jest.fn().mockResolvedValue({ data: { currentRules: { text: "<p>123 testing</p>" } } }),
      ...handlers
    }

    mockClient.setRequestHandler(
      CURRENT_RULES_QUERY,
      requestHandlers.currentRulesHandler
    )

    wrapper = shallowMount(index, { apolloProvider, localVue, vuetify })
  }

  afterEach(() => {
    wrapper.destroy()
    mockClient = null
    apolloProvider = null
  })

  test("shows 'no rules yet' when there is no currentRules", async () => {
    createComponent({
      currentRulesHandler: jest.fn().mockResolvedValue({ data: { currentRules: null } })
    })
    await wrapper.vm.$nextTick()
    await wrapper.vm.$nextTick()
    expect(wrapper.text()).toContain("As regras desta temporada ainda nÃ£o foram cadastradas")
  })

  test("shows 'loading'", async () => {
    createComponent()

    await wrapper.vm.$nextTick()
    expect(wrapper.text()).toContain("Carregando...")
  })

  test("shows rules text", async () => {
    createComponent()

    await wrapper.vm.$nextTick()
    expect(wrapper.vm.$apollo.queries.currentRules).toBeTruthy()
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
    await wrapper.vm.$nextTick()
    expect(wrapper.text()).toContain("123 testing")
  })
})
