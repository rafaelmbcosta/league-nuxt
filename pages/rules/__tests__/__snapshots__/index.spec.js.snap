// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index shows rules text 1`] = `
<div>
  <!---->
   
  <div>
    
    Carregando...
  
  </div>
</div>
`;
